### PR TITLE
Fixes #19431 - retry command on session expiry

### PR DESCRIPTION
**Pull request by tstrachota**
Generic mechanism for retrying commands when they return `EX_RETRY` either from method `execute` or from exception handler.

### CODE REVIEW

**Comment by tstrachota** [comment]
Requires https://github.com/theforeman/hammer-cli-foreman/pull/309 to work for session expiry.

**Comment by tstrachota** [comment]
@ares @mbacovsky I guess you would be interested in taking a look at the code.

File: lib/hammer_cli/exit_codes.rb [hunk 1]
```diff
@@ -20,4 +20,5 @@ module HammerCLI
   # non POSIX codes
   EX_NOT_FOUND    = 128     # resource was not found
   EX_UNAUTHORIZED = 129     # authorization failed
+  EX_RETRY        = 666     # retry the command
```
Comment (ares) [inline lib/hammer_cli/exit_codes.rb]: what a number :-)

**Comment by ares** [comment]
Works very well :+1: Few ideas for later improvements - it might not log the unauhtorized 401 as error log level when ran with `-d`, especially the trace indicates some problem, even though 4xx HTTP code means it's client error. I can also imagine it would suggest the login from last time so when I hit enter on username prompt it would use the previous login, but those are really just nitpick. +1000 for this super cool feature, could we get this in some upcoming release? :-)

**Comment by tstrachota** [comment]
Merged, thanks for testing it @ares

**Pull request merged**